src/python/pants/fs/archive.py

```diff
@@ -163,9 +163,17 @@ def _invoke_xz(self, xz_input_file):
     env = {
       # Isolate the path so we know we're using our provided version of xz.
       'PATH': xz_bin_dir,
-      # Only allow our xz's lib directory to resolve the liblzma.{so,dylib} dependency at runtime.
-      'LD_LIBRARY_PATH': self._xz_library_path,
     }
+
+    # Only allow our xz's lib directory to resolve the liblzma.{so,dylib} dependency at runtime.
+    normalized_os_name = get_normalized_os_name()
+    if normalized_os_name == 'darwin':
+      env['DYLD_FALLBACK_LIBRARY_PATH'] = self._xz_library_path
```

Comment:
I think this should probably actually be:
`env['DYLD_LIBRARY_PATH'] = '{}:{}'.format(self.._xz_library_path, env.get('DYLD_LIBRARY_PATH', ''))`

Two important differences:
1. Set the main path not the fallback
2. Preserve existing values; they may be important

(The one below should probably also set the existing path)

But I think it would be better for us to make sure the `xz` we release to pantsbinaries runs out of the box (either by linking differently, or with a wrapper script), rather than patching this up in pants...

Reply:
So there is a very useful method `get_joined_path()` I added to `contextutil.py` to do the path operations you describe. However, I believe pantsbuild/binaries#71 lets us remove these lines. I will make the changes here and wait for that PR to be merged.

Reply:
Added a wrapper script to the `xz` released to pants-binaries in pantsbuild/binaries#72! Thanks, that was an very useful suggestion.